pkg/plugins/optional/grafana/alphav1/plugin.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package alphav1
+
+import (
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+	cfgv3 "sigs.k8s.io/kubebuilder/v3/pkg/config/v3"
+	"sigs.k8s.io/kubebuilder/v3/pkg/model/stage"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugins"
+)
+
+const pluginName = "grafana.common." + plugins.DefaultNameQualifier
+
+var (
+	pluginVersion            = plugin.Version{Number: 1, Stage: stage.Alpha}
+	supportedProjectVersions = []config.Version{cfgv3.Version}
+)
+
+// Plugin implements the plugin.Full interface
+type Plugin struct {
+	initSubcommand
+
+	// TODO:
+	// editSubcommand
+}
+
+var (
+	// FIXME: Following line is just respect to similar pattern appears in other plugin implementation.
+	//        Not sure about the reason. (From @Kavinjsir, 2022-06-11)
+	_ plugin.Init = Plugin{}
+)
+
+// Name returns the name of the plugin
+func (Plugin) Name() string { return pluginName }
+
+// Version returns the version of the plugin
+func (Plugin) Version() plugin.Version { return pluginVersion }
+
+// SupportedProjectVersions returns an array with all project versions supported by the plugin
+func (Plugin) SupportedProjectVersions() []config.Version { return supportedProjectVersions }
+
+// GetInitSubcommand will return the subcommand which is responsible for initializing and common scaffolding
+func (p Plugin) GetInitSubcommand() plugin.InitSubcommand { return &p.initSubcommand }
+
+// TODO
+// GetEditSubcommand will return the subcommand which is responsible for editing the scaffold of the project
+// func (p Plugin) GetEditSubcommand() plugin.EditSubcommand { return &p.editSubcommand }
```

Comment:
This file is == https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/v3/plugin.go
However, we only implement init and edit which means in the structure we have only both
```go
// Plugin implements the plugin.Full interface
type Plugin struct {
	initSubcommand
	editSubcommand
}
```

Also, here: (we need to change )

```go

const pluginName = "base." + golang.DefaultNameQualifier

var (
	pluginVersion            = plugin.Version{Number: 3}
	supportedProjectVersions = []config.Version{cfgv3.Version}
)

```

with

```
const pluginName = "grafana." + plugins.DefaultNameQualifier

var (
	pluginVersion            = plugin.Version{Number: 1, Stage: stage.Alpha}
	supportedProjectVersions = []config.Version{cfgv3.Version}
	pluginKey                = plugin.KeyFor(Plugin{})
)
```

So that we are defining the key of your plugin as version

Reply:
Thx for the guidance! @camilamacedo86 

Noticed that in the above, we have:
```go
var (
        ...
	pluginKey                = plugin.KeyFor(Plugin{})
)
```

Wondering the intention to have `pluginKey` defined here, how would we like to use it?

I notice that [in the implementation of cli.WithPlugin](https://github.com/kubernetes-sigs/kubebuilder/blob/v3.4.1/pkg/cli/options.go#L61), `plugin.KeyFor` will be called for each plugin passed in.

So wondering if it is necessary to have such value here?

Reply:
I think this one is not required but you can check by looking also in the other plugins.